### PR TITLE
fix(search): dont search non string aggregations

### PR DESCRIPTION
**Pull request by patricebender**
with this change:

- functions are **not** searched per default
- function expressions which are common string aggregations are searched per default → having clause
- function expressions which are cast to string are searched → having clause (while a custom `substring` will also be available to the `where` clause, we can make sure that it would also work with a custom aggregation function)

### CODE REVIEW

File: db-service/test/cqn4sql/search.test.js [hunk 1]
```diff
@@ -133,4 +150,55 @@ describe('Replace attribute search by search predicate', () => {
...
+      } group by title
+      `
+
+    query.SELECT.search = [{ val: 'x' } ]
+
+    expect(JSON.parse(JSON.stringify(cqn4sql(query, model)))).to.deep.equal(CQL`
+      SELECT from bookshop.Books as Books {
+        Books.ID,
+        substring(Books.stock) as searchRelevantViaCast: cds.String,
+      } group by Books.title having search(substring(Books.stock), 'x')`)
```
[the remark this answers is not in the document]
Comment (patricebender) [reply db-service/test/cqn4sql/search.test.js:202]: Yes, column expressions are evaluated before having, hence we could also use the reference here. I think it is a good idea 👍

[the remark this answers is not in the document]
Comment (patricebender) [reply db-service/test/cqn4sql/search.test.js]: implemented and seems to work:

```js
> q
Query {
  SELECT: {
    from: { ref: [ 'sap.capire.bookshop.Books' ] },
    columns: [
      {
        args: [ { ref: [ 'title' ] } ],
        as: 'firstInAlphabet',
        func: 'MIN'
      }
    ],
    groupBy: [ { ref: [ 'title' ] } ],
    search: [ { val: 'Cat' } ]
  }
}
> q.forSQL()
Query {
  SELECT: {
    from: { ref: [ 'sap.capire.bookshop.Books' ], as: 'Books' },
    columns: [
      {
        func: 'MIN',
        args: [ { ref: [ 'Books', 'title' ] } ],
        as: 'firstInAlphabet'
      }
    ],
    groupBy: [ { ref: [ 'Books', 'title' ] } ],
    search: [ { val: 'Cat' } ],
    having: [
      {
        func: 'search',
        args: [ { ref: [ 'firstInAlphabet' ] }, { val: 'Cat' } ]
      }
    ]
  }
}
> await q
[ { firstInAlphabet: 'Catweazle' } ]
```

[the remark this answers is not in the document]
Comment (patricebender) [reply db-service/test/cqn4sql/search.test.js:202]: it seems like standard sql disallows this: https://dba.stackexchange.com/questions/281438/why-does-an-alias-with-a-having-clause-not-exist-in-postgresql

File: db-service/lib/search.js [hunk 2]
```diff
@@ -131,24 +137,38 @@ const computeColumnsToBeSearched = (cqn, entity = { __searchableColumns: [] }, a
           return
         }
 
-        toBeSearched.push({ func: column.func, args: column.args })
+        // only strings can be searched)
```
Comment (patricebender) [inline db-service/lib/search.js:140]:
```suggestion
        // only strings can be searched
```

**Review by johannes-vogel — APPROVED** [review]

**Pull request merged**